TelemetryProcessors/QuickPulseTelemetryProcessor.ts

```diff
@@ -0,0 +1,27 @@
+import Contracts = require("../Declarations/Contracts");
+import Logging = require("../Library/Logging");
+import QuickPulseStateManager = require("../Library/QuickPulseStateManager")
+import AutoCollectPerformance = require("../AutoCollection/Performance");
+
+export function quickPulseTelemetryProcessor(envelope: Contracts.Envelope, client?: QuickPulseStateManager): boolean {
+    if (client) {
+        client.addDocument(envelope);
+
+        // Increment rate counters
+        switch (envelope.data.baseType) {
+            case "ExceptionData":
+                AutoCollectPerformance.countException();
+                break;
+            case "RequestData":
+                // These are already autocounted by HttpRequest.
+                // Note: Not currently counting manual trackRequest calls
+                // here to avoid affecting existing autocollection metrics
+                break;
+            case "RemoteDependencyData":
+                const baseData: Contracts.DependencyTelemetry = (envelope.data as any).baseData;
+                AutoCollectPerformance.countDependency(baseData.duration, baseData.success);
```

Comment:
Why not have the RDD autocollector just do this internally, like we do for request counting?

Reply:
My thinking was that I also want to send 3rd party tracks as documents, and I don't think those pass through HttpDependencies at all? I could add this logic to `trackDependency`, but since I have to do it for exceptions as well, I kept it all here instead.

Reply:
Will only quickpulse ever need these stats? Or are they also standard metrics we weren't reporting before? If its only quickpulse, keeping it here sounds good.

Reply:
Assuming you mean trackMetrics I added in `Performance.ts`. There may be extra ones that we aren't sending right now. I was planning on adding those in a separate PR since it'll change the default behavior of our perf telemetry.
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/01f38eb50ffc05bd4e60c28754e8f5672356a307/Src/PerformanceCollector/Perf.Shared/PerformanceCollectorModule.cs#L154-L171